server/router/api/v1/resource_service.go

```diff
@@ -355,7 +355,7 @@ func SaveResourceBlob(ctx context.Context, s *store.Store, create *store.Resourc
 		if err != nil {
 			return errors.Wrap(err, "Failed to upload via s3 client")
 		}
-		presignURL, err := s3Client.PresignGetObject(ctx, key)
+		presignURL, err := s3Client.PresignGetObject(ctx, key, s3Config)
```

Comment:
We should check custom domain here instead of inside `s3Client.PresignGetObject`.

Reply:
There are two options: 1) Change the function name to s3Client.GetObject, 2) Check the custom domain here. I updated the code to use option 2, but maybe option 1 is better